### ISSUE TITLE
global name 'basic_sentiment_analysis' is not defined.

### ISSUE THREAD

**Issue by linas** [opencog/opencog#2285]
I get this message on startup:

```
ERROR: In procedure opencog-extension:
ERROR: Throw to key `C++-EXCEPTION' with args `("cog-evaluate!" "Python error in call_sentiment_parse: global name 'basic_sentiment_analysis' is not defined. (/home/linas/src/atomspace/opencog/cython/PythonEval.cc:802)")'.
```

**Comment by linas** [comment]
probably due to a dependency on nltk.

The basic-sentiment-analysis code should be moved to a distinct directory, and converted into a module.

The component that actually needs this should perform the analysis, instead of jamming it into nlp-parse (which does not actually need this function).

The nltk tokenizer should not be used, since relex already provides a tokenization.

**Comment by linas** [comment]
In general r2l-parse should perform only the r2l-parsing and nothing else. If other analysis is needed, it should be done in a distinct step, instead of getting tangled into r2l.

**Comment by ngeiswei** [comment]
@linas you probably need to install pyyaml, as indicated in chatbot/ready4sentiment
```
sudo pip install pyyaml
```

**Comment by AmeBel** [comment]
For working setup,
```
sudo pip install -U pyyaml nltk
sudo python -m nltk.downloader -d /usr/local/share/nltk_data punkt averaged_perceptron_tagger
```

**Comment by linas** [comment]
yeah, this needs to be refactored so that the code can still work without installing all these extra dependencies.

**Comment by linas** [comment]
fixed with work-around #2401
